pages/manage/[pubkey].tsx

```diff
@@ -3,69 +3,47 @@ import type { NextPage } from 'next'
 import { useRouter } from 'next/router'
 import { useEffect } from 'react'
 import WalletDetails from '../../components/WalletDetails'
+import LoadingSpinner from '../../components/LoadingSpinner'
+import { useAppSelector } from '../../hooks/useAppSelector'
+import { selectCluster } from '../../redux/features/wallet/walletSlice'
+import useSWR from 'swr'
 
-export const getServerSideProps = async (context: any) => {
-  const pubKey = context.params.pubkey
-
-  // get the Wallet details here
-  const wallet = {
-    pubKey: pubKey,
-    name: 'MY WALLEEEEET',
-    members: [
-      {
-        address: 'D4QYC..ioi',
-        addedOn: 'April 15, 2022',
-        shares: 23,
-      },
-      {
-        address: 'D4QYC..ioi',
-        addedOn: 'April 15, 2022',
-        shares: 23,
-      },
-      {
-        address: 'D4QYC..ioi',
-        addedOn: 'April 15, 2022',
-        shares: 23,
-      },
-      {
-        address: 'D4QYC...ioi',
-        addedOn: 'April 15, 2022',
-        shares: 23,
-      },
-      {
-        address: 'D4QYC...ioi',
-        addedOn: 'April 15, 2022',
-        shares: 23,
-      },
-    ],
-    authority: 'D4QYC...ioi',
-    shares: 100,
-    model: 'Wallet membership',
-    acceptSPL: true,
-    pubKeySPL: '2msQ3eha1rvhSuwCPJ213oVgwtu3FSgxBXS6RPrzsDWB',
-  }
-  return {
-    props: { wallet: wallet },
+const fetcher = (key: string) => {
+  if (key) {
+    return fetch(key).then((res) => res.json())
   }
 }
 
-interface Props {
-  wallet: any
-}
-
-const WalletDetailsPage: NextPage<Props> = ({ wallet }) => {
-  const { connected } = useWallet()
+const WalletDetailsPage: NextPage = () => {
   const router = useRouter()
+  const pubKey = router.query.pubkey
+  const { connected } = useWallet()
+  const cluster = useAppSelector(selectCluster)
 
   useEffect(() => {
     if (router.isReady && !connected) {
       router.replace('/')
     }
   }, [router, connected])
 
+  const endpoint = pubKey ? `/api/wallets/${pubKey}?cluster=${cluster}` : ''
+  const { data, error } = useSWR(endpoint, fetcher)
+
+  if (error) {
+    return <p className="text-center">Failed to load wallet</p>
+  }
+
+  if (!data) {
+    return (
+      <div className="flex flex-row justify-center items-center">
+        <LoadingSpinner />
+      </div>
+    )
+  }
+
   return (
     <div className="container mx-auto px-6 sm:px-0 gap-10 flex flex-col justify-center items-center my-10">
-      <WalletDetails wallet={wallet} />
+      <WalletDetails wallet={data} />
```

Comment:
The API returns the wallet data in the following format:

```typescript
{
  found: boolean,
  wallet: {
    name: string,
    pubkey: string,
    authority: string,
    memberShipType: string,
    acceptSPL: boolean,
    totalShares: number,
    cluster: string,
    validated: boolean
  },
  members: {
    memberPubkey: string,
    createdAt: Date,
    shareCount: number,
    cluster: string,
    walletPubkey: string
  }[]
}
```

You would check what format the `WalletDetails` component expects the data to be in and map it into that format. Also don't forget to check the `MembersTable` component to see how you'd map the `members` field. Notice that the API returns the `members` array separate from the `wallet` object, so you'd have to include it into the `wallet` object.

Reply:
Resolving this.